sufia-models/app/actors/sufia/generic_file/actor.rb

```diff
@@ -102,11 +107,22 @@ def save_and_record_committer
         return false unless generic_file.save
       rescue RSolr::Error::Http => error
         ActiveFedora::Base.logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Caught RSOLR error #{error.inspect}"
+        Sufia.config.upload_logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Error saving #{generic_file.title}. Caught RSOLR error #{error.inspect}"
+        save_tries += 1
+        # fail for good if the tries is greater than 3
+        raise error if save_tries >= 3
+        sleep 0.01
+        retry
+      rescue Ldp::Gone => error
+        Sufia.config.upload_logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Error saving #{generic_file.title}. Caught LDP error #{error.inspect}"
         save_tries += 1
         # fail for good if the tries is greater than 3
         raise error if save_tries >= 3
         sleep 0.01
         retry
+      rescue StandardError => error
+        Sufia.config.upload_logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Error saving #{generic_file.title}. Caught Exception error #{error.inspect}"
```

Comment:
The changes in this file bring up a couple related questions:

1. Why doesn't the `StandardError` rescue block also deal with `save_tries` and the other code in the above blocks?
2. Seeing all the repetition in the blocks... maybe we should just rescue all exceptions and treat them all the same way.